src/utils/utilities.go

```diff
@@ -29,15 +27,52 @@ func UnitToDivider(unit pb.RateLimitResponse_RateLimit_Unit) int64 {
 	panic("should not get here")
 }
 
-func CalculateReset(currentLimit *pb.RateLimitResponse_RateLimit, timeSource TimeSource) *duration.Duration {
-	sec := UnitToDivider(currentLimit.Unit)
-	now := timeSource.UnixNow()
-	return &duration.Duration{Seconds: sec - now%sec}
+func MaxInt(a int, b int) int {
+	if a > b {
+		return a
+	}
+	return b
 }
 
-func Max(a uint32, b uint32) uint32 {
+func MaxInt64(a int64, b int64) int64 {
 	if a > b {
 		return a
 	}
 	return b
 }
+
+func MinInt64(a int64, b int64) int64 {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func MaxUint32(a uint32, b uint32) uint32 {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+func NanosecondsToSeconds(nanoseconds int64) int64 {
```

Comment:
looks like those functions belong to time file?

Reply:
actually, I move out all function that doesn't belong to struct (method) / creating the struct itself/used by the struct to utilities.go, wdyt?

Reply:
that is one way of approaching it, i would still put time related utilities in time file